### PR TITLE
Initial sample

### PR DESCRIPTION
**Pull request by derisen**
How to setup:

1. git clone `https://github.com/Azure-Samples/ms-identity-javascript-nodejs-tutorial.git`
2. switch to branch `derisen`
4. Locate the individual sample folder and run `npm install & npm start`.

Things to review for:

- auto-registration scripts
- manual registration instructions
- login/logout/call in different browsers
- login/logout/call in private mode
- any errors in console
- README text/grammar etc.

### CODE REVIEW

File: 1-Authentication/1-sign-in/README.md [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+Locate the root of the sample folder. Then:
+
+```console
+    npm start
+```
+
+## Explore the sample
+
+1. Open your browser and navigate to `http://localhost:4000`.
+1. Select the **sign-in** button on the top right corner.
```
Comment (kalyankrishna1) [inline 1-Authentication/1-sign-in/README.md:152]: >corner [](http://example.com/codeflow?start=50&length=6)

Also mention clicking the top right button to check id token claims

File: 1-Authentication/1-sign-in/AppCreationScripts/Configure.ps1 [hunk 2]
```diff
@@ -0,0 +1,197 @@
+[CmdletBinding()]
+param(
```
Comment (kalyankrishna1) [inline 1-Authentication/1-sign-in/AppCreationScripts/Configure.ps1:2]: >param [](http://example.com/codeflow?start=0&length=5)

note that this script has issues with , in appsettings.json

File: 1-Authentication/1-sign-in/README.md [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+
+> :warning: To avoid path length limitations on Windows, we recommend cloning into a directory near the root of your drive.
+
+### Step 2: Install project dependencies
+
+Locate the root of the sample folder. Then:
+
+```console
+    cd 1-Authentication\1-sign-in
+    npm install
```
Comment (kalyankrishna1) [inline 1-Authentication/1-sign-in/README.md:65]: >npm install [](http://example.com/codeflow?start=4&length=11)

note that msal-express module does not install correctly after this

File: 1-Authentication/1-sign-in/README.md [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+    /**
+     * Validates the id token for a set of claims
+     * @param {Object} idTokenClaims: decoded id token claims
+     */
+    validateIdToken = (idTokenClaims): boolean => {
+        const now = Math.round((new Date()).getTime() / 1000); // in UNIX format
+
+        const checkAudience = idTokenClaims["aud"] === this.msalConfig.auth.clientId ? true : false;
+        const checkTimestamp = idTokenClaims["iat"] <= now && idTokenClaims["exp"] >= now ? true : false;
+        const checkTenant = (this.appSettings.policies && !idTokenClaims["tid"]) || idTokenClaims["tid"] === this.appSettings.credentials.tenantId ? true : false;
```
Comment (kalyankrishna1) [inline 1-Authentication/1-sign-in/README.md:384]: >checkTenant [](http://example.com/codeflow?start=14&length=11)

would this interfere for MT apps?

File: 1-Authentication/2-sign-in-b2c/README.md [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+
+#### Configure the client app (msal-node-webapp) to use your app registration
+
+Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `appSettings.json` file.
+1. Find the key `clientId` and replace the existing value with the application ID (clientId) of `msal-node-webapp` app copied from the Azure portal.
+1. Find the key `tenantId` and replace the existing value with your Azure AD tenant ID.
```
Comment (kalyankrishna1) [inline 1-Authentication/2-sign-in-b2c/README.md:111]: >Azure AD tenant ID [](http://example.com/codeflow?start=68&length=18)

Azure AD B2C tenant id?

File: 1-Authentication/2-sign-in-b2c/README.md [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+### Register the client app (msal-node-webapp)
+
+1. Navigate to the [Azure portal](https://portal.azure.com) and select the **Azure AD B2C** service.
+1. Select the **App Registrations** blade on the left, then select **New registration**.
+1. In the **Register an application page** that appears, enter your application's registration information:
+   - In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `msal-node-webapp`.
+   - Under **Supported account types**, select **Accounts in any identity provider or organizational directory (for authenticating users with user flows)**.
+   - In the **Redirect URI (optional)** section, select **Web** in the combo-box and enter the following redirect URI: `http://localhost:4000/redirect`.
+1. Select **Register** to create the application.
+1. In the app's registration screen, find and note the **Application (client) ID**. You use this value in your app's configuration file(s) later in your code.
```
Comment (kalyankrishna1) [inline 1-Authentication/2-sign-in-b2c/README.md:94]: >. [](http://example.com/codeflow?start=157&length=1)

For B2C, add a line.. 
"also note the 'Directory (tenant) ID' value"

File: 1-Authentication/2-sign-in-b2c/README.md [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+ 1. [Running the sample](#running-the-sample)
+ 1. [Explore the sample](#explore-the-sample)
+ 1. [About the code](#about-the-code)
+ 1. [More information](#more-information)
+ 1. [Community Help and Support](#community-help-and-support)
+ 1. [Contributing](#contributing)
+
+## Overview
+
+This sample demonstrates a Node.js & Express web application that authenticates users against Azure AD, with the help of [Microsoft Authentication Library for Node.js](https://aka.ms/msalnode) (MSAL Node). In doing so, it illustrates authentication concepts such as [OpenID scopes](https://docs.microsoft.com/azure/active-directory/develop/v2-permissions-and-consent#openid-connect-scopes), [ID Tokens](https://docs.microsoft.com/azure/active-directory-b2c/openid-connect), [ID Token validation](https://docs.microsoft.com/azure/active-directory-b2c/openid-connect#validate-the-id-token), [user-flows](https://docs.microsoft.com/azure/active-directory-b2c/user-flow-overview) and more.
```
Comment (idg-sam) [inline 1-Authentication/2-sign-in-b2c/README.md:18]: remove: against
Add: Azure AD **B2C**

File: 2-Authorization-I/1-call-graph/README.md [hunk 8]
```diff
@@ -0,0 +1,401 @@
...
+1. Find the key `redirectUri` and replace the existing value with the Redirect URI for `msal-node-webapp`. (by default `http://localhost:4000/redirect`).
+1. Find the key `postLogoutRedirectUri` and replace the existing value with the base address of `msal-node-webapp` (by default `http://localhost:4000/`).
+
+The rest of the **key-value** pairs are for resources/APIs that you would like to call. They are set as **default**, but you can modify them as you wish:
+
+```json
+    "nameOfYourResource": {
+        "callingPageRoute": "/<route_where_this_resource_will_be_called_from>",
+        "endpoint": "<uri_coordinates_of_the_resource>",
+        "scopes": ["scope1_of_the_resource", "scope1_of_the_resource", "..."]
```
Comment (idg-sam) [inline 2-Authorization-I/1-call-graph/README.md:162]:
```suggestion
        "scopes": ["scope1_of_the_resource", "scope2_of_the_resource", "..."]
```

File: 1-Authentication/1-sign-in/README.md (hunk 1 above)
Comment (derisen) [reply 1-Authentication/1-sign-in/README.md:152]: done!

File: 1-Authentication/2-sign-in-b2c/README.md (hunk 5 above)
Comment (derisen) [reply 1-Authentication/2-sign-in-b2c/README.md:111]: done

File: 1-Authentication/1-sign-in/README.md (hunk 3 above)
Comment (derisen) [reply 1-Authentication/1-sign-in/README.md:65]: problem resolved. forgot to check-in the `dist` folder previously

File: 1-Authentication/1-sign-in/AppCreationScripts/Configure.ps1 (hunk 2 above)
Comment (derisen) [reply 1-Authentication/1-sign-in/AppCreationScripts/Configure.ps1:2]: Will address this in codegenerator

File: 1-Authentication/1-sign-in/README.md (hunk 4 above)
Comment (derisen) [reply 1-Authentication/1-sign-in/README.md:385]: Revised this to handle multitenant case

**Pull request merged**